daemon/graphdriver/driver.go

```diff
@@ -85,6 +86,26 @@ type Driver interface {
 	DiffSize(id, parent string) (size int64, err error)
 }
 
+// CustomImageAccessor is an interface to allow callers to query the object for
+// any implementation specific image information that might need to be inserted into
+// the graph.
+type CustomImageAccessor interface {
+	// GetCustomImages returns a json string with metadata describing the images
+	// known to this implementation. The data is a json encoded CustomImageInfoList.
+	GetCustomImages() (string, error)
+}
+
+type CustomImageInfo struct {
+	Name        string
+	Version     string
+	Path        string
+	Size        int64
+	CreatedTime time.Time
+}
+type CustomImageInfoList struct {
```

Comment:
`CustomImageInfos` -- not sure if this type is worth exporting since it only provides serialization definition. This can just be anonymously defined as the `json.Marshal` call site.